query/src/main/scala/filodb/query/exec/PeriodicSamplesMapper.scala

```diff
@@ -118,9 +118,9 @@ final case class PeriodicSamplesMapper(start: Long,
   // Transform source double or long to double schema
   override def schema(source: ResultSchema): ResultSchema = {
     // Special treatment for downsampled gauge schema - return regular timestamp/value
-    if (source.columns == Schemas.dsGauge.dataInfos) {
+    if (functionId.contains(AvgWithSumAndCountOverTime)) {
```

Comment:
Are you sure this is only needed for that function ID?  It seems to be all other downsampled gauge needs the transform too. Otherwise any other function on downsampled data just copies the original schema, which would contain multiple columns right?

Reply:
Source would be only selected columns (timestamp and one of the cols) which would be right. AvgWithSumAndCountOverTime is the only function that would have selected two columns.